langserver/workspace_refs.go

```diff
@@ -149,7 +161,14 @@ func (h *LangHandler) workspaceRefsTypecheck(ctx context.Context, bctx *build.Co
 			}
 			return bpkg, nil
 		},
-		AfterTypeCheck: afterTypeCheck,
+		AfterTypeCheck: func(pkg *loader.PackageInfo, files []*ast.File) {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
```

Comment:
you can simplify your done checks. Just do
```
if ctx.Err() != nil {
  return
}
```

this applies to all your other uses as well.